### PR TITLE
Verify Guardians

### PR DESCRIPTION
**Pull request by garyray-k**

### CODE REVIEW

**Comment by garyray-k** [comment]
Screenshot 2023-05-02 at 8 59 43 PM
Screenshot 2023-05-02 at 8 59 34 PM

Click through to Verify Guardians.
The verify code should be copyable and the form should not be valid unless all guardian codes are entered.

This does not interact with the config API at all and is just the UI template.

File: guardian-ui/src/components/VerifyGuardians.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            <Input
+              value={clipValue}
+              readOnly={true}
+              onChange={(e) => {
+                setValue(e.target.value);
+              }}
+              mr={2}
+            />
+            <Button onClick={onCopy}>{hasCopied ? 'Copied!' : 'Copy'}</Button>
+          </Flex>
```
Comment (wbobeirne) [inline guardian-ui/src/components/VerifyGuardians.tsx:102]: I have a `CopyInput` component in a branch that has the Host's invite code, I'll be sure to re-use that here when I put that PR up.

File: guardian-ui/src/components/VerifyGuardians.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        </FormControl>
+      </FormGroup>
+      <FormGroup>
+        <FormControl>
+          <FormGroupHeading title="Enter each guardian's verification code" />
+          {otherVerificationCodes}
+        </FormControl>
+      </FormGroup>
+      <div>
+        {!isValid && <FormLabel>All codes must be entered</FormLabel>}
```
Comment (wbobeirne) [inline guardian-ui/src/components/VerifyGuardians.tsx:112]: Disabling button, marking fields required, _and_ showing a message might be a little belt-and-suspenders. We should come up with a consistent UI pattern for how we handle forms with required fields, but the designs didn't have a persistent message like this for the other forms with required fields.

File: guardian-ui/src/components/FormGroupHeading.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  icon?: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
+  title: React.ReactNode;
+}> = ({ icon, title }) => (
+  <Flex align='center' justify='start' mb={icon ? -3 : 3}>
+    {icon && <Icon width='20px' height='20px' mr={2} as={icon} />}
+    <Heading fontSize='md' lineHeight='20px' fontWeight='700'>
+      {title}
+    </Heading>
+  </Flex>
+);
```
Comment (wbobeirne) [inline guardian-ui/src/components/FormGroupHeading.tsx:14]: I was putting components that might be candidates for a reusable UI library in the `ui/` folder. Not sure if these make sense there, but they could if we have a lot of similar looking forms.

**Review by wbobeirne — APPROVED** [review]

File: guardian-ui/src/components/FormGroupHeading.tsx (hunk 3 above)
Comment (garyray-k) [reply guardian-ui/src/components/FormGroupHeading.tsx:14]: Ahh. Ok. It wasn't clear to me how the folder structure was organized. I'll keep this in mind going forward. This component would fall under the `ui` folder in this case.

File: guardian-ui/src/components/VerifyGuardians.tsx (hunk 2 above)
Comment (garyray-k) [reply guardian-ui/src/components/VerifyGuardians.tsx:112]: Good call. I'll solicit the design channel in Discord for feedback.

**Comment by wbobeirne** [comment]
> Self reminder: Need to check that codes are not duplicated in form validation.

Just had a chat with @jkitman who's going to re-do the API for codes so that the validation is all client-side. This will give us a lot of flexibility to manage the UX of code validation. I'd hold off on doing anything on that front until his work is up.

**Pull request merged**